### PR TITLE
Anko - Change Profile layout to anko

### PR DESCRIPTION
**Pull request by at-haingo**
#### Related issues (Required)
- Task #108

#### Description (Required)
- [x] Change xml to anko layout

### CODE REVIEW

File: app/src/main/java/vn/asiantech/way/ui/register/RegisterActivityUI.kt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                    rightMargin = R.dimen.register_screen_avatar_margin
+                }
+            }.lparams {
+                topMargin = dip(56)
+                centerHorizontally()
+            }
+        }
+    }
+
+    inline fun ViewManager.circleImageView(theme: Int = 0, init: CircleImageView.() -> Unit): CircleImageView {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/register/RegisterActivityUI.kt]: WARNING: Public functions require documentation.

Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/register/RegisterActivityUI.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/register/RegisterActivityUI.kt [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        return ankoView({ CircleImageView(it) }, theme, init)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/register/RegisterActivityUI.kt]: WARNING: Checks whether files end with a line separator.

Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/register/RegisterActivityUI.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/register/RegisterActivityUI.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                }
+                circleImageView {
+
+                }.lparams {
+                    gravity = Gravity.END
+                    backgroundDrawable = ContextCompat.getDrawable(context, R.drawable.ic_profile_camera)
+                    rightMargin = R.dimen.register_screen_avatar_margin
+                }
+            }.lparams {
+                topMargin = dip(56)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/register/RegisterActivityUI.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/register/RegisterActivityUI.kt [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+                centerInParent()
+            }
+        }
+    }
+
+    /*
+     * Add circleImageView library
+     */
+    private inline fun ViewManager.circleImageView(theme: Int = 0, init: CircleImageView.() -> Unit):
+            CircleImageView {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/register/RegisterActivityUI.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/register/CountryAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+                }
+            }
+            itemView.tag = CountryHolder(itemView, imgFlag, tvTel)
+            return itemView
+        }
+    }
+
+    var onItemClick: (Country) -> Unit = {}
+
+
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/register/CountryAdapter.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/register/CountryAdapter.kt [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    /**
+     * Holder Country
+     */
+    inner class CountryHolder(itemView: View, private val imgFlag: ImageView,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/register/CountryAdapter.kt]: WARNING: Unexpected indentation

File: app/src/main/java/vn/asiantech/way/ui/register/CountryAdapter.kt [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    companion object {
+        private const val IMG_FLAG_ID = 1001
+        private const val IMG_FLAG_WIDTH = 36
+        private const val IMG_FLAG_HEIGHT = 28
+        private const val TV_TEL_MARGIN = 10
+    }
+
+    val mFlags: HashMap<String, Bitmap> = getFlagMap()
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = ItemCountryUI()
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/register/CountryAdapter.kt]: ERROR: Cast operator throws an exception if the cast is not possible.

File: app/src/main/java/vn/asiantech/way/ui/register/CountryAdapter.kt [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+import java.util.*
+
+/**
+ * Adapter of RecyclerView Country.
+ * Created by haingoq on 27/11/2017.
+ */
+class CountryAdapter(private val context: Context, private val countries: List<Country>) :
+        RecyclerView.Adapter<CountryAdapter.CountryHolder>() {
+    companion object {
+        private const val IMG_FLAG_ID = 1001
```
Comment (at-tienhoang) [inline app/src/main/java/vn/asiantech/way/ui/register/CountryAdapter.kt]: Please change to `ID_IMG_FLAG`

File: app/src/main/java/vn/asiantech/way/ui/register/CountryAdapter.kt [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+        private const val IMG_FLAG_WIDTH = 36
+        private const val IMG_FLAG_HEIGHT = 28
+        private const val TV_TEL_MARGIN = 10
```
Comment (at-tienhoang) [inline app/src/main/java/vn/asiantech/way/ui/register/CountryAdapter.kt]: Move 3 lines to dimens xml

File: app/src/main/java/vn/asiantech/way/ui/register/CountryAdapter.kt [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+            tvTel.text = getCountryName(isoCode.toLowerCase())
+        }
+    }
+
+    private fun getCountryName(isoCode: String): String {
+        val locale = Locale(Locale.getDefault().displayLanguage, isoCode)
+        return locale.displayCountry.trim()
+    }
+
+    private fun getFlagMap(): HashMap<String, Bitmap> {
```
Comment (at-tienhoang) [inline app/src/main/java/vn/asiantech/way/ui/register/CountryAdapter.kt]: Move to activity.java

File: app/src/main/java/vn/asiantech/way/ui/register/RegisterActivityUI.kt [hunk 11]
```diff
@@ -0,0 +1,223 @@
...
+/**
+ * Anko layout for RegisterActivity
+ * Created by haingoq on 27/11/2017.
+ */
+class RegisterActivityUI(private val countryAdapter: CountryAdapter) : AnkoComponent<RegisterActivity> {
+    companion object {
+        // ID of view
+        private const val FR_AVATAR_ID = 101
+        private const val TV_DESCRIPTION_ID = 102
+        private const val RL_INFORMATION_ID = 103
```
Comment (at-tienhoang) [inline app/src/main/java/vn/asiantech/way/ui/register/RegisterActivityUI.kt]: Change names and values follow convention

File: app/src/main/java/vn/asiantech/way/ui/register/CountryAdapter.kt [hunk 12]
```diff
@@ -0,0 +1,92 @@
+package vn.asiantech.way.ui.register
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/register/CountryAdapter.kt]: WARNING: Unused import

File: app/src/main/java/vn/asiantech/way/ui/register/CountryAdapter.kt [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+import android.support.v7.widget.RecyclerView
+import android.view.Gravity
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import org.jetbrains.anko.*
+import org.jetbrains.anko.sdk25.coroutines.onClick
+import vn.asiantech.way.R
+import vn.asiantech.way.data.model.Country
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/register/CountryAdapter.kt]: WARNING: Unused import

File: app/src/main/java/vn/asiantech/way/ui/register/CountryAdapter.kt [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+import java.io.InputStream
+import java.util.*
+import kotlin.collections.HashMap
+
+/**
+ * Adapter of RecyclerView Country.
+ * Created by haingoq on 27/11/2017.
+ */
+class CountryAdapter(private val context: Context, private val countries: List<Country>,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/register/CountryAdapter.kt]: WARNING: Unexpected indentation

File: app/src/main/java/vn/asiantech/way/ui/register/RegisterActivity.kt [hunk 15]
```diff
@@ -1,452 +1,37 @@
...
-
-    private fun saveLoginStatus(login: Boolean) {
-        val editor = mSharedPreferences.edit()
-        editor.putBoolean(KEY_LOGIN, login)
-        editor.apply()
+        RegisterActivityUI(CountryAdapter(createCountries(), HashMap<String, Bitmap>(), "VietNam"))
+                .setContentView(this)
+    }
+
+    private fun createCountries(): List<Country> {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/register/RegisterActivity.kt]: Please add `// TODO`

File: app/src/main/java/vn/asiantech/way/ui/register/RegisterActivityUI.kt [hunk 16]
```diff
@@ -0,0 +1,214 @@
...
+        private const val ID_FR_AVATAR = 1001
+        private const val ID_TV_DESCRIPTION = 1002
+        private const val ID_RL_INFORMATION = 1003
+        private const val ID_BTN_SAVE = 1004
+        private const val ID_TV_SKIP = 1005
+        private const val ID_EDT_NAME = 1006
+        private const val ID_VIEW_LINE = 1007
+    }
+
+    internal lateinit var dialog: DialogInterface
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/register/RegisterActivityUI.kt]: change variable name to `dialogInterface`

File: app/src/main/java/vn/asiantech/way/ui/register/RegisterActivityUI.kt [hunk 17]
```diff
@@ -0,0 +1,214 @@
...
+    internal lateinit var tvCancel: TextView
+    internal lateinit var progressBar: ProgressBar
+
+    override fun createView(ui: AnkoContext<RegisterActivity>) = with(ui) {
+        relativeLayout {
+            lparams(matchParent, matchParent)
+            frAvatar = frameLayout {
+                id = ID_FR_AVATAR
+
+                circleImageView {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/register/RegisterActivityUI.kt]: Please change  `circleImageView( R.drawable.ic_default_avatar) {`

File: app/src/main/java/vn/asiantech/way/ui/register/RegisterActivityUI.kt [hunk 18]
```diff
@@ -0,0 +1,214 @@
...
+                isEnabled = false
+            }.lparams(matchParent, dimen(R.dimen.register_screen_save_button_height)) {
+                val margin = dip(dimen(R.dimen.register_screen_btn_register_margin))
+                below(ID_RL_INFORMATION)
+                leftMargin = margin
+                topMargin = margin
+                rightMargin = margin
+            }
+
+            tvSkip = textView(resources.getString(R.string.register_skip)) {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/register/RegisterActivityUI.kt]: Change this line to `tvSkip = textView(R.string.register_skip) {`

File: app/src/main/java/vn/asiantech/way/ui/register/RegisterActivity.kt [hunk 19]
```diff
@@ -0,0 +1,33 @@
+package vn.asiantech.way.ui.register
+
+import android.graphics.Bitmap
+import android.os.Bundle
+import org.jetbrains.anko.setContentView
+import vn.asiantech.way.R
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/register/RegisterActivity.kt]: WARNING: Unused import

**Pull request merged**